pac4j-core/src/main/java/org/pac4j/core/client/Clients.java

```diff
@@ -104,7 +105,10 @@ protected void internalInit() {
      */
     public Client findClient(final WebContext context) {
         final String name = context.getRequestParameter(this.clientNameParameter);
-        CommonHelper.assertNotBlank("name", name);
+		if (name == null && defaultClient != null) {
```

Comment:
basically if on callback url there is no clientName specified you can configure in a default client., It's not related to resource changes.. I'm just not sure how to make that change separate at this point.